lib/ui/text.dart

```diff
@@ -933,6 +933,64 @@ class FontFeature {
   String toString() => "FontFeature('$feature', $value)";
 }
 
+/// An axis tag and value that can be used to customize variable fonts.
+///
+/// Some fonts are variable fonts that can generate a range of different
+/// font faces by altering the values of the font's design axes.
+///
+/// See https://docs.microsoft.com/en-us/typography/opentype/spec/otvaroverview
+class FontVariation {
+  /// Creates a [FontVariation] object, which can be added to a [TextStyle] to
+  /// change the variable attributes of a font.
+  ///
+  /// `axis` is the four-character tag that identifies the design axis.
+  /// These tags are specified by font formats such as OpenType.
+  /// See https://docs.microsoft.com/en-us/typography/opentype/spec/dvaraxisreg
+  ///
+  /// `value` is the value that the axis will be set to. The behavior
+  /// depends on how the font implements the axis.
+  const FontVariation(
+    this.axis,
+    this.value,
+  ) : assert(axis != null),
+      assert(axis.length == 4, 'Axis tag must be exactly four characters long.'),
+      assert(value != null);
+
+  /// The tag that identifies the design axis.  Must consist of 4 ASCII
+  /// characters.
+  final String axis;
+
+  /// The value assigned to this design axis.
+  ///
+  /// The range of usable values depends on the specification of the axis.
+  final double value;
```

Comment:
Does it make sense to document these? Are there commonly used values? Can we document more about the acceptable ranges of each character?

Reply:
The usable values depend on the definition of each axis.  The user will need to consult the documentation for the font and the font format (such as the OpenType spec linked above)